server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -300,17 +306,22 @@ private CompletionStage<ConditionalResponse> replaceIfUnmodifiedAfterGet(Advance
             .thenApply(replaced -> new ConditionalResponse(replaced, entry));
    }
 
-   private void handleConditionalResponse(HotRodHeader header, ConditionalResponse response, Throwable throwable) {
+   private void handleConditionalResponse(HotRodHeader header, ConditionalResponse response, Throwable throwable,
+                                          InfinispanSpan<ConditionalResponse> span) {
       if (throwable != null) {
          writeException(header, throwable);
+         span.accept(null, throwable);
       } else if (response.result) {
          assert response.entry != null;
          writeSuccess(header, response.entry);
+         span.accept(response, null);
       } else {
          if (response.entry == null) {
             writeNotExist(header);
+            span.accept(null, null);
          } else {
             writeNotExecuted(header, response.entry);
+            span.accept(response, null);
          }
```

Comment:
or ignore my comment above and write a try-finally block in all methods like this:

```java
try {
  if (throwable != null) {
    // writeException
    span.recordException(throwable)
  } else {
    // other logic
  }
} finally {
  span.complete()
}
```

Reply:
We could do that... hrmm

Reply:
I will change to this, it will be much better. Since we don't care about the value, it is much cleaner.